base/commands/list/list_set.go

```diff
@@ -6,54 +6,65 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/hazelcast/hazelcast-commandline-client/base"
 	"github.com/hazelcast/hazelcast-commandline-client/clc"
+	"github.com/hazelcast/hazelcast-commandline-client/clc/cmd"
 	. "github.com/hazelcast/hazelcast-commandline-client/internal/check"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/output"
 	"github.com/hazelcast/hazelcast-commandline-client/internal/plug"
 	"github.com/hazelcast/hazelcast-commandline-client/internal/proto/codec"
 )
 
 type ListSetCommand struct{}
 
+func (mc *ListSetCommand) Unwrappable() {}
+
 func (mc *ListSetCommand) Init(cc plug.InitContext) error {
 	cc.SetCommandUsage("set")
 	help := "Set a value at the given index in the list"
 	cc.SetCommandHelp(help, help)
 	addValueTypeFlag(cc)
 	cc.AddInt64Arg(argIndex, argTitleIndex)
-	cc.AddStringArg(argValue, argTitleValue)
+	cc.AddStringArg(base.ArgValue, base.ArgTitleValue)
 	return nil
 }
 
 func (mc *ListSetCommand) Exec(ctx context.Context, ec plug.ExecContext) error {
-	name := ec.Props().GetString(listFlagName)
-	ci, err := ec.ClientInternal(ctx)
-	if err != nil {
-		return err
-	}
-	// get the list just to ensure the corresponding proxy is created
-	if _, err := ec.Props().GetBlocking(listPropertyName); err != nil {
-		return err
-	}
+	name := ec.Props().GetString(base.FlagName)
 	index := ec.GetInt64Arg(argIndex)
-	valueStr := ec.GetStringArg(argValue)
-	vd, err := makeValueData(ec, ci, valueStr)
-	if err != nil {
-		return err
-	}
-	pid, err := stringToPartitionID(ci, name)
-	if err != nil {
-		return err
-	}
-	_, stop, err := ec.ExecuteBlocking(ctx, func(ctx context.Context, sp clc.Spinner) (any, error) {
+	valueStr := ec.GetStringArg(base.ArgValue)
+	rowV, stop, err := ec.ExecuteBlocking(ctx, func(ctx context.Context, sp clc.Spinner) (any, error) {
+		ci, err := cmd.ClientInternal(ctx, ec, sp)
+		if err != nil {
+			return nil, err
+		}
+		// get the list just to ensure the corresponding proxy is created
+		_, err = getList(ctx, ec, sp)
+		if err != nil {
+			return nil, err
+		}
+		vd, err := makeValueData(ec, ci, valueStr)
+		if err != nil {
+			return nil, err
+		}
+		pid, err := stringToPartitionID(ci, name)
+		if err != nil {
+			return nil, err
+		}
 		sp.SetText(fmt.Sprintf("Setting the value of the list %s", name))
 		req := codec.EncodeListSetRequest(name, int32(index), vd)
-		return ci.InvokeOnPartition(ctx, req, pid, nil)
+		resp, err := ci.InvokeOnPartition(ctx, req, pid, nil)
+		if err != nil {
+			return nil, err
+		}
+		data := codec.DecodeListSetResponse(resp)
+		return convertDataToRow(ci, data, ec.Props().GetBool(base.FlagShowType))
 	})
 	if err != nil {
 		return err
 	}
 	stop()
-	return nil
+	return ec.AddOutputRows(ctx, rowV.(output.Row))
```

Comment:
We are printing the last set value as follows
```
Value 
last-val  
```
We are also using the same title for every return type. I find this a bit confusing, what do you think?

Reply:
> same title for every return type

Do you mean every data structure?

Reply:
Not every data structure, I realized this function `convertDataToRow(ci *hazelcast.ClientInternal, data hazelcast.Data, showType bool)` is only used once in `list set`. 

What I meant is we are printing the correct title for every other command and for this one we are printing the last value with the `Value` title which is a bit misleading.

Reply:
Changed the titles at:
* https://github.com/hazelcast/hazelcast-commandline-client/pull/372/commits/e9cfedad237d7557064057f20e20803dad14f0c9
*  https://github.com/hazelcast/hazelcast-commandline-client/pull/372/commits/9965a3088c4445fbc02d47e620525d8a0eb320c3